src/OrchardCore.Modules/OrchardCore.Contents/Deployment/AddToDeploymentPlan/AddToDeploymentPlanController.cs

```diff
@@ -74,7 +74,15 @@ await _authorizationService.AuthorizeAsync(User, OrchardCore.Deployment.CommonPe
                 return Forbid();
             }
 
-            var step = (ContentItemDeploymentStep)_factories.FirstOrDefault(x => x.Name == nameof(ContentItemDeploymentStep)).Create();
+            var stepFactory = _factories.FirstOrDefault(x => x.Name == nameof(ContentItemDeploymentStep));
+
+            if (stepFactory == null)
+            {
+                return BadRequest();
+            }
+
+            var step = (ContentItemDeploymentStep)stepFactory.Create();
```

Comment:
```suggestion
            var step = (ContentItemDeploymentStep)stepFactory.Create();
```
```suggestion
            var step = (ContentItemDeploymentStep)stepFactory;
            
            if (step != null)
            {
                stepFactory.Create();
            }
```

Reply:
Why?!! we already checked the factory in line 79

Reply:
Ok, because of the FirstOrDefault but it is not given that an explicit casting will return a value generally.